### DISCUSSION TITLE
New layout.tsx after Nextjs 15 update

### DISCUSSION THREAD

**Discussion by luiscaceresd** [amannn/next-intl#1502] [Q&A] [answered]
Hi guys, I'm having a difficult time upgrading to nextjs 15. I already implemented the suggested changes here: https://next-intl-docs.vercel.app/blog/next-intl-3-22#await-request-locale 
But I'm still getting this error: 

`Error: Route "/[locale]" used `params.locale`. `params` should be awaited before using its properties. Learn more: https://nextjs.org/docs/messages/sync-dynamic-apis
    at createParamsAccessError (turbopack://[project]/node_modules/next/dist/src/server/request/params.ts:467:9)
    at getMessage (turbopack://[project]/node_modules/next/dist/src/server/create-deduped-by-callsite-server-error-logger.ts:46:20)
    at warnForSyncAccess (turbopack://[project]/node_modules/next/dist/src/server/request/params.ts:445:4)
    at syncIODev (turbopack://[project]/node_modules/next/dist/src/server/request/params.ts:403:10)
    at param (turbopack://[project]/node_modules/next/src/server/server-utils.ts:74:25)
    at pathname (turbopack://[project]/node_modules/next/src/lib/metadata/get-metadata-route.ts:40:39)
    at tree.children.metadata.openGraph (/home/kira/02-WORK/00-lezgo/lezgo/.next/server/chunks/ssr/node_modules_cf91f8._.js:114:241)
    at imageModule (turbopack://[project]/node_modules/next/dist/src/lib/metadata/resolve-metadata.ts:381:27)
    at Array.map (<anonymous>)
    at map (turbopack://[project]/node_modules/next/dist/src/lib/metadata/resolve-metadata.ts:379:58)
    at collectStaticImagesFiles (turbopack://[project]/node_modules/next/dist/src/lib/metadata/resolve-metadata.ts:399:4)
    at resolveStaticMetadata (turbopack://[project]/node_modules/next/dist/src/lib/metadata/resolve-metadata.ts:449:36)
    at async resolveMetadataItemsImpl (turbopack://[project]/node_modules/next/dist/src/lib/metadata/resolve-metadata.ts:548:2)
    at async resolveMetadataItemsImpl (turbopack://[project]/node_modules/next/dist/src/lib/metadata/resolve-metadata.ts:562:4)`

I ran the nextjs 15 codemod and it updated my layout to this:

```tsx
import Header from "@/components/navbar/Header";
import { cn } from '@/lib/utils';
import { NextIntlClientProvider } from 'next-intl';
import { getMessages } from 'next-intl/server';
import { Inter } from 'next/font/google';
import "./globals.css";
import { Providers } from "./providers";
import Footer from "./Footer";
import Script from "next/script";
import { GoogleTagManager } from '@next/third-parties/google'
import localfont from "next/font/local"
import { getTenants } from "@/utils/prisma/getTenants";
import { CSPostHogProvider } from "./_providers/CSPosstHogProvider";
import { routing } from "@/i18n/routing";
import { ReactNode } from "react";

const fontHeading = Inter({
  subsets: ['latin'],
  display: 'swap',
  variable: '--font-heading',
})

const fontBody = localfont({
  src: [
    {
      path: "../../public/fonts/omnes/omnes-regular.ttf",
      weight: "400",
    },
    {
      path: "../../public/fonts/omnes/omnes-medium.ttf",
      weight: "500",
    },
    {
      path: "../../public/fonts/omnes/omnes-semiBold.ttf",
      weight: "600",
    },
  ],
  variable: '--font-body',
});

const lotaGrotesque = localfont({
  src: [
    {
      path: "../../public/fonts/lota-grotesque/LotaGrotesque-Regular.otf",
      weight: "400",
    },
    {
      path: "../../public/fonts/lota-grotesque/LotaGrotesque-SemiBold.otf",
      weight: "600",
    },
    {
      path: "../../public/fonts/lota-grotesque/LotaGrotesque-Bold.otf",
      weight: "700",
    },
  ],
  variable: '--font-lota',
});

const defaultUrl = process.env.VERCEL_URL
  ? `https://${process.env.VERCEL_URL}`
  : "http://localhost:3000";

export const metadata = {
  metadataBase: new URL(defaultUrl),
  title: "Lezgo",
  description: "The fastest way to rent cars",
};

export default async function LocaleLayout(
  props: {
    children: React.ReactNode;
    params: Promise<{ locale: string }>;
  }
) {
  const params = await props.params;

  const {
    locale
  } = params;

  const {
    children
  } = props;

  const messages = await getMessages();
  const tenants = await getTenants();

  return (
    <html lang={locale} data-theme="light">
      <GoogleTagManager gtmId="GTM-NGNQ5L4W" />
      <CSPostHogProvider>
        <body className={cn(
          'antialiased',
          'overflow-x-hidden',
          fontHeading.variable,
          fontBody.variable,
          lotaGrotesque.variable,
        )}>
          <NextIntlClientProvider
            messages={messages}
            formats={{
              dateTime: {
                short: {
                  timeZone: 'UTC',
                  month: 'short',
                  day: 'numeric',
                },
                long: {
                  timeZone: 'UTC',
                  weekday: 'short',
                  month: 'short',
                  day: 'numeric',
                  year: 'numeric',
                },
              }
            }}
          >
            <Providers initialTenants={tenants}>
              <div className="min-h-screen w-screen flex flex-col overflow-hidden">
                <Header />
                <main className="min-h-[94dvh] flex flex-col items-center flex-1 w-screen gap-8 mb-20">
                  <div className="flex-1 flex flex-col items-center w-screen gap-4">
                    {children}
                  </div>
                </main>
                <Footer />
              </div>
            </Providers>
          </NextIntlClientProvider>
        </body>
      </CSPostHogProvider>
    </html>
  );
}
```

So I'm not sure why I'm getting this

**Comment by amannn** [comment]
Have you also switched to [`createNavigation`](https://next-intl-docs.vercel.app/blog/next-intl-3-22#create-navigation)? (See the note on Next.js 15 in the linked section)

**Comment by luiscaceresd** [comment]
Yeah, this is my routing.ts

```ts
import {defineRouting} from 'next-intl/routing';
import {createNavigation} from 'next-intl/navigation';
 
export const routing = defineRouting({
  locales: ['en', 'es', 'nl'],
  defaultLocale: 'en',
  alternateLinks: false
});

export type Locale = (typeof routing.locales)[number];

export const {Link, getPathname, redirect, usePathname, useRouter} =
  createNavigation(routing);
  ```

**Reply by amannn** [reply]
Hmm, can't see an issue with that. In your stack trace, there are a few mentions of "metadata". Maybe you're reading `params.locale` yourself somewhere and need to update that!

**Comment by emirxmertoglu** [comment]
Try this instead,
```ts
type Params = Promise<{ locale: string }>;

export default async function LocaleLayout({
  children,
  params,
}: {
  children: React.ReactNode;
  params: Params;
}) {
  const { locale } = await params;
```

**Comment by luiscaceresd** [comment, marked as answer]
Guys I was going crazy over this, the issue was this: https://stackoverflow.com/questions/79143162/route-locale-used-params-locale-params-should-be-awaited-before-using

![image](https://github.com/user-attachments/assets/c7913596-efbb-4c05-a096-69635c7b5928)

I can finally migrate to nextjs 15 without any warnings lol
